docs/02-app/02-api-reference/04-functions/after.mdx

```diff
@@ -0,0 +1,70 @@
+---
+title: after
+description: API Reference for the after function.
+version: RC
+---
+
+`after()` allows you to schedule work to be executed after a response is finished. This is useful for tasks and other side effects that should not block the response, such as logging and analytics.
+
+It can be used in [Server Components](/docs/app/building-your-application/rendering/server-components) (including [`generateMetadata`](https://nextjs.org/docs/app/api-reference/functions/generate-metadata)), [Server Actions](/docs/app/building-your-application/data-fetching/server-actions-and-mutations), [Route Handlers](/docs/app/building-your-application/routing/route-handlers), and [Middleware](/docs/app/building-your-application/routing/middleware).
+
+The function accepts a callback that will be executed after the response is finished:
+
+```tsx filename="app/layout.tsx switcher
+import { after } from 'next/server'
+import { log } from '@/app/utils'
+
+export default function Layout({ children }: { children: React.ReactNode }) {
+  after(() => {
+    // Execute after the layout is rendered and sent to the user
+    log()
+  })
+  return <>{children}</>
+}
+```
+
+```jsx filename="app/layout.js switcher
+import { after } from 'next/server'
+import { log } from '@/app/utils'
+
+export default function Layout({ children }) {
+  after(() => {
+    // Execute after the layout is rendered and sent to the user
+    log()
+  })
+  return <>{children}</>
+}
+```
+
+> **Good to know**:
+>
+> - `after` will be executed even if the response didn't complete successfully. Including when an error is thrown or when `notFound` or `redirect` is called.
+> - `after` is a [dynamic function](/docs/app/building-your-application/rendering/server-components#dynamic-functions) that will opt a route into dynamic rendering. This behavior can be overridden with the [`export dynamic = "force-static"`](/docs/app/api-reference/file-conventions/route-segment-config#dynamic) segment config.
+> - You can use React `cache` to deduplicate functions called inside `after`.
+> - [`cookies`](/docs/app/api-reference/functions/cookies) cannot be set inside `after` since the response has already been sent.
+> - `after` can be nested inside other `after` calls.
```

Comment:
How do the callbacks run? Id like to see an example of nesting

Reply:
Did a masking tape fix until I can think of a compelling use case. 😅